diff --git a/dist/hooks.js b/dist/hooks.js
index cba4eed326e8c0b9f157812953c992ee29e5a71e..e3214bbf7f082afe2a2fa90e05ba6fe74651aff8 100644
--- a/dist/hooks.js
+++ b/dist/hooks.js
@@ -6,7 +6,11 @@ async function promisifyHooks(hooks, args) {
     if (!hooks || !Array.isArray(hooks)) {
         return Promise.resolve();
     }
-    await Promise.all(hooks.map(hookFn => (0, util_1.promisify)(hookFn).apply(promisifyHooks, args)));
+    await Promise.all(hooks.map(hookFn => {
+        const result = hookFn.apply(promisifyHooks, args);
+        if (result && typeof result.then === 'function') return result;
+        return (0, util_1.promisify)(hookFn).apply(promisifyHooks, args);
+    }));
 }
 exports.promisifyHooks = promisifyHooks;
 /**
